service/internal/obsconsumer/logs.go

```diff
@@ -29,18 +29,22 @@ func NewLogs(cons consumer.Logs, itemCounter, sizeCounter metric.Int64Counter, o
 		opt(&o)
 	}
 
+	consumerSet := Settings{
+		ItemCounter: set.ItemCounter,
+		SizeCounter: set.SizeCounter,
+		Logger:      set.Logger.With(zapFields(o.staticDataPointAttributes)...),
```

Comment:
@jade-guiton-dd here I use `Logger.With` to add the `otelcol.pipeline.id` and any other attributes, I am not sure if I should use [`componentattribute.ZapLoggerWithAttributes`](https://github.com/open-telemetry/opentelemetry-collector/blob/aef8b1c3d26c062bc31e4f96c54f5f2a03b304c6/internal/telemetry/componentattribute/logger_zap.go#L139-L144) instead. What do you think?